### PR TITLE
Collapsed the long app descriptions

### PR DESCRIPTION
**Pull request by mihaistate03**
Implemented a method to collapse by default all descriptions that are longer
than a specified value. Created a button for extending the description and
one for collapsing it.
Addressed issue #537 

Signed-off-by: Mihai State <mihai.state@codethink.co.uk>

### CODE REVIEW

**Comment by razzeee** [comment]
I actually think we should not do this

**Comment by ifeeltiredboss** [comment]
Maybe it should be discussed? Is description capped in any way or it can be as long as the developer wants?
I know that Flathub is not obliged to follow others but:
1. Google collapses that https://play.google.com/store/apps/details?id=org.telegram.messenger
2. Apple collapses that https://apps.apple.com/pl/app/telegram/id686449807
3. MS does too https://www.microsoft.com/en-us/p/telegram-messenger/9wzdncrdzhs0

**Comment by razzeee** [comment]
I'm not aware of a limit and there is no mention of that in the spec
https://freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-description

Btw, how weird MS implementation is, having the description twice on that page.

File: frontend/src/components/application/Details.tsx [hunk 1]
```diff
@@ -214,10 +233,28 @@ const Details: FunctionComponent<Props> = ({
...
+              }}
             />
           </div>
+          {scrollHeight > collapsedHeight && (
+            <button
+              className="w-full rounded-tl-none rounded-tr-none rounded-bl-none rounded-br-none border-t py-3 px-0 font-semibold hover:cursor-pointer hover:bg-colorHighlight dark:border-zinc-600"
+              {...getToggleProps()}
+            >
+              {isExpanded ? (
+                <span>{t("Se less")}</span>
```
Comment (razzeee) [inline frontend/src/components/application/Details.tsx:252]: Translation keys should be lowercase with no spaces.

This one also has a typo

**Comment by mihaistate03** [comment]
Addressed the suggestion from the review regarding the translation keys in the latest commit.

**Comment by razzeee** [comment]
I think the button needs padding and we can probably get rid of the vertical line.

Here's gnome sofwares take 

image

**Comment by mihaistate03** [comment]
image
Is it ok this button style?

**Comment by razzeee** [comment]
Yeah, feel free to push that, I can iterate on the details somewhen later

File: backend/pyproject.toml [hunk 2]
```diff
@@ -8,7 +8,7 @@ license = "Apache-2.0"
 [tool.poetry.dependencies]
 python = "^3.10"
 fastapi = "^0.80.0"
-uvicorn = "^0.18.2"
+uvicorn = "^0.18.3"
```
Comment (razzeee) [inline backend/pyproject.toml:11]: Backend change shouldn't be needed, possibly a rebase error?

File: frontend/public/locales/en/common.json [hunk 3]
```diff
@@ -249,5 +249,9 @@
   "token-redeem-placeholder": "Enter an ownership token",
   "open-menu": "Open Menu",
   "open-user-menu": "Open User Menu",
-  "your-profile": "Your Profile"
+  "your-profile": "Your Profile",
+  "see-full-description": "See full description",
```
Comment (razzeee) [inline frontend/public/locales/en/common.json:253]: I don't think this is used

File: frontend/public/locales/en/common.json [hunk 4]
```diff
@@ -249,5 +249,9 @@
...
+  "see-less": "See less",
```
Comment (razzeee) [inline frontend/public/locales/en/common.json:254]: Unused

File: backend/pyproject.toml (hunk 2 above)
Comment (mihaistate03) [reply backend/pyproject.toml:11]: It is indeed an error, I haven`t make any changes to this file.

Comment (razzeee) [reply backend/pyproject.toml:11]: You probably messed up during rebase, might want to be abit more careful there. I'll go ahead and fix this for now :)

**Comment by razzeee** [comment]
@mihaistate03 I took the liberty to modify this quiet heavily, so you might want to have another look at the source. Anyway, I'll merge it and maybe look into replacing these expandable elements with headlessUI disclosures.

**Pull request merged**